python/lsst/ctrl/bps/wms/pegasus/pegasus_service.py

```diff
@@ -0,0 +1,400 @@
+            with open(pegout, "w") as pegfh:
+                pegfh.write(f"Command: {cmd}\n\n")
+                process = subprocess.run(shlex.split(cmd), shell=False, stdout=pegfh,
+                                         stderr=subprocess.STDOUT, check=False)
+                if process.returncode != 0:
+                    print(f"Error trying to generate Pegasus files.  See {pegout}.")
+                    raise RuntimeError(f"pegasus-plan exited with non-zero exit code ({process.returncode})")
+
+            # Grab run id from pegasus-plan output and save
+            with open(pegout, "r") as pegfh:
+                for line in pegfh:
+                    match = re.search(r"pegasus-run\s+(\S+)", line)
+                    if match:
+                        self.run_id = match.group(1)
+                        break
+
+        # Hack - Using profile in sites.xml doesn't add run attributes to DAG submission
+        # file. So adding them here:
+        if run_attr is not None:
+            subname = f'{self.run_id}/{self.name}-0.dag.condor.sub'
+            shutil.copyfile(subname, subname + ".orig")
+            with open(subname + ".orig", "r") as infh:
+                with open(subname, "w") as outfh:
+                    for line in infh:
+                        if line.strip() == "queue":
+                            htc_write_attribs(outfh, run_attr)
+                            htc_write_attribs(outfh, {"bps_job_label": "DAG"})
+                        outfh.write(line)
+
+    def _write_properties_file(self, out_prefix, filenames):
+        """Write Pegasus Properties File
+        """
+        properties = f"{self.name}_pegasus.properties"
+        if out_prefix is not None:
+            properties = os.path.join(out_prefix, properties)
+        with open(properties, "w") as outfh:
+            outfh.write("# This tells Pegasus where to find the Site Catalog.\n")
+            outfh.write(f"pegasus.catalog.site.file={filenames['sites']}\n")
+
+            outfh.write("# This tells Pegasus where to find the Replica Catalog.\n")
+            outfh.write(f"pegasus.catalog.replica.file={filenames['replica']}\n")
+
+            outfh.write("# This tells Pegasus where to find the Transformation Catalog.\n")
+            outfh.write("pegasus.catalog.transformation=Text\n")
+            outfh.write(f"pegasus.catalog.transformation.file={filenames['transformation']}\n")
+
+            outfh.write("# Run Pegasus in shared file system mode.\n")
+            outfh.write("pegasus.data.configuration=sharedfs\n")
+
+            outfh.write("# Make Pegasus use links instead of transferring files.\n")
+            outfh.write("pegasus.transfer.*.impl=Transfer\n")
+            outfh.write("pegasus.transfer.links=true\n")
+
+            # outfh.write("# Use a timestamp as a name for the submit directory.\n")
+            # outfh.write("pegasus.dir.useTimestamp=true\n")
+            # outfh.write("pegasus.dir.storage.deep=true\n")
```

Comment:
remove commented code?

Reply:
I also prefer `print("My message", file=outfh)` in general over using `write()`